### PR TITLE
Fix profiling tool reading collectionAccumulator

### PR DESCRIPTION
**Pull request by tgravescs**
The profiling tool can fail to process an event log if it has custom task accumulables of type CollectionAccumulator, this just changes it to catch the exception and ignore those accumulables.

This is a minimal fix for https://github.com/NVIDIA/spark-rapids/issues/5091 but I want to leave that open and handle these better.

Signed-off-by: Thomas Graves <tgraves@nvidia.com>

### CODE REVIEW

**Comment by tgravescs** [comment]
build

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/EventsProcessor.scala [hunk 1]
```diff
@@ -404,8 +404,8 @@ class EventsProcessor(app: ApplicationInfo) extends EventProcessorBase[Applicati
         app.accumIdToStageId.put(res._2.id, event.stageInfo.stageId)
         arrBuf += thisMetric
       } catch {
-        case e: ClassCastException =>
-          logWarning("ClassCastException when parsing accumulables for task " +
+        case e: Exception =>
```
Comment (gerashegalov) [inline tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/EventsProcessor.scala:407]: should we catch exceptions more exhaustively with matching on `Throwable`?

Comment (tgravescs) [reply tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/EventsProcessor.scala:407]: if it's an Error, I think we just want to fail as that could be something more than just related to the accumulable, what I actually probably should do is make this match on NonFatal(e)

**Comment by tgravescs** [comment]
build

**Review by revans2 — APPROVED** [review]

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/EventsProcessor.scala (hunk 1 above)
Comment (gerashegalov) [reply tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/EventsProcessor.scala]: +1 for NonFatal, it's consistent with similar usages in our code base

**Review by gerashegalov — APPROVED** [review]

**Pull request merged**